### PR TITLE
EUSS Demand Flexibility Measures emission reduction / decarbonization objective

### PR DESCRIPTION
**Pull request by JieXiong9119**
Pull request overview
---------------------

Code review for updated demand flexibility measures with emission reduction objective and tests

- lighting control load shed (updated)
- thermostat control load shed (updated)
- dispatch schedule generation (updated)

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [x] Upgrade Measures
 - [ ] Reporting Measures
 - [ ] Postprocessing

Author pull request checklist:

 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Added tests for new measures
 - [x] Updated measure .xml(s)
 - [ ] Register values added to `comstock_column_definitions.csv`
 - [x] Both `options_lookup.tsv` files updated
 - [x] 10k+ test run
 - [ ] Change documentation written
 - [x] Measure documentation written
 - [x] ComStock documentation updated
 - [x] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [x] Added 'See ComStock License' language to first two lines of each code file
 - [ ] Implements corresponding measure tests and indexing path in `test/measure_tests.txt` or/and `test/resource_measure_tests.txt`
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

**Comment by JieXiong9119** [comment]
This update is based on the df lighting measure branch and PR #199, so most of the comments of that PR would be applicable here. I suggest reviewers hold your review on this one until relevant comments are addressed in the other PR.

File: resources/measures/upgrade_df_thermostat_control_load_shed/measure.rb [hunk 1]
```diff
@@ -371,19 +438,37 @@ def applicable_thermostats(model)
...
+      "SmallOffice",
+      "MediumOffice",
       "LargeOffice",
       "OfL",
       "OfS",
-      "Office"
+      "Office",
+      "Warehouse",
+      "PrimarySchool",
+      "SecondarySchool"
```
Comment (JanghyunJK) [inline resources/measures/upgrade_df_thermostat_control_load_shed/measure.rb]: @JieXiong9119 don't we have other building type names for warehouse/school in CA? similar to OfL/OfS?

Comment (JanghyunJK) [reply resources/measures/upgrade_df_thermostat_control_load_shed/measure.rb]: maybe this: https://github.com/NREL/ComStock/blob/1f5866c4dd9a499df2523fc20f5a017a3349036b/resources/measures/set_hvac_template/resources/deer_building_types.rb#L10-L36

File: resources/Gemfile
Comment (ChristopherCaradonna) [inline resources/Gemfile:1]: verify if this should be included.

File: .rubocop.yml
Comment (ChristopherCaradonna) [inline .rubocop.yml:1]: verify if this should be included.

**Review by eringold — APPROVED** [review]

**Pull request merged**